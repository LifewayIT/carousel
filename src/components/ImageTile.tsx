import React, { ButtonHTMLAttributes, ReactElement, ReactNode } from 'react';
import styled from 'styled-components';
import { device } from '../utils/styleguide';

const TileButton = styled.button`
  display: block;
  padding: 0;
  border: 0 none;
  background: none;

  border-radius: 4px;
  box-shadow: 0 2px 8px -2px rgb(45 55 72 / 25%);

  &:hover {
    cursor: pointer;
  }

  @media (prefers-reduced-motion: no-preference) {
    transition: all .1s ease-out;

    &:hover {
      transform: scale(1.12);
    }
  }

  @media (hover: none), (hover: on-demand) {
    &:hover {
      transform: unset;
    }
  }

  .selected & {
    box-shadow:
      0 0 0 4px #38b2ac,
      0 2px 8px 2px rgb(45 55 72 / 25%);
  }
`;

const TileImage = styled.img`
  height: 128px;
  border-radius: 4px;
  vertical-align: top;

  @media ${device._600} {
    height: 144px;
  }

  @media ${device._900} {
    height: 176px;
  }

  @media ${device._1024} {
    height: 200px;
  }
`;

type Props = {
  /** the source url of the image */
  src: string;
  /** the alt text for the image */
  alt: string;
  /** child nodes to render inside of the button (helpful for banners, markers, etc) */
  children?: ReactNode;
} & ButtonHTMLAttributes<HTMLButtonElement>;

export const ImageTile = ({ src, alt, children, ...props }: Props): ReactElement => {
  return (
    <TileButton {...props}>
      <TileImage src={src} alt={alt} />
      {children}
    </TileButton>
  );
};
