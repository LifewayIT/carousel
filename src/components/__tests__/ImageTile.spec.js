import React from 'react';
import { render } from '@testing-library/react';
import { ImageTile } from '../ImageTile';

test('matches snapshot', () => {
  const { container } = render(
    <ImageTile className="test-class" src="/src" alt="image label">
      <div id="children" />
    </ImageTile>
  );

  expect(container.firstChild).toMatchSnapshot();
});
