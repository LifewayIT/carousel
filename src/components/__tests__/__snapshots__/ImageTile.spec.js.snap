// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  class="lwc-image-tile lwc-tile test-class"
>
  <img
    alt="image label"
    class="lwc-tile-image"
    src="/src"
  />
  {children}
</button>
`;
